### ISSUE TITLE
Mac 'os' grain is broken in version 2015.8.0

### ISSUE THREAD

**Issue by sparrc** [saltstack/salt#27488]

**Comment by sparrc** [comment]
After upgrading to 2015.8.0, Mac OSX is no longer able to get the os via salt grains, and instead it just gets `proxy` (it used to get `MacOS`). It looks like this might be because of a new dmesg requirement?

```
$ salt-call grains.item os --local
[WARNING ] Although 'dmesg' was found in path, the current user cannot execute it. Grains output might not be accurate.
local:
    ----------
    os:
        proxy
```

**Comment by sparrc** [comment]
PS: Mac OSX has dmesg, but must be read via `sudo`. Salt on mac probably shouldn't require `sudo` because OSX has a very strong separation between admin and user capabilities.

**Comment by jfindlay** [comment]
@sparrc, thanks for the report.  @cro, do you have any ideas on this one?

**Comment by cro** [comment]
This should be fixed in 2015.8.1.  In the meantime you can replace salt/utils/__init__.py with the one from the 2015.8 branch, the fix is in there.
